internal/validation/config.go

```diff
@@ -0,0 +1,119 @@
+package validation
+
+import (
+	"context"
+	"fmt"
+	"github.com/speakeasy-api/openapi-generation/v2/pkg/generate"
+	sdkGenConfig "github.com/speakeasy-api/sdk-gen-config"
+	"github.com/speakeasy-api/speakeasy-client-sdk-go/v3/pkg/models/shared"
+	"github.com/speakeasy-api/speakeasy-core/events"
+	"github.com/speakeasy-api/speakeasy/internal/log"
+	"regexp"
+	"strings"
+)
+
+// ValidateConfigAndPrintErrors validates the generation config for a target and prints any errors
+// Returns an error if the config is invalid, nil otherwise
+func ValidateConfigAndPrintErrors(ctx context.Context, target string, cfg *sdkGenConfig.Config, publishingEnabled bool) error {
```

Comment:
This name is a little confusing. It says validate and print errors. But it looks like we return an error and are not swallowing it

Reply:
It prints the errors nicely, then returns if there are any errors
I agree its a bit wonky, but not sure of anything better. It is invoked differently now so maybe that helps